doc/Manuals/ug-sec_x_road_security_hardening.md

```diff
@@ -0,0 +1,165 @@
+The method of adding a new password policy varies significantly depending on your operating system, existing PAM configuration and authentication protocol.
+
+For instruction on how to add password policies, please refer to your operating system's official documentation or customer support.
+
+#### 2.2.1 Considerations and risks
+
+In a strong password, it is advisable to have at least 16 characters at minimum. You can also add complexity requirements, such as numbers and special characters, but these requirements can make the passwords more difficult for users to remember. Further additional measures could be to add commonly known passwords into a blocklist.
+
+### 2.3 Ensuring User Account Security
+
+Users of the web application are created by creating operating-system-level users. This means that a user can access the web application and the underlying operating system with the same credentials. 
+Therefore, if user accounts in the web application were compromised, the attacker could use those credentials to log into the server via SSH if credential-based logging in is not disabled.
+
+To harden the user account security, make sure that users are not allowed to access the server via SSH by default. The users needing SSH access are granted those rights separately.
+
+Create a user group in which users are allowed to connect to the server via SSH while all other users are denied. Add the following line to `/etc/ssh/sshd_config`:
+
+    AllowGroups <group_to_allow>
+
+Restart the SSH service:
+
+    sudo systemctl restart sshd
+
+It is also recommended to disable SSH password login and allow key-based authentication only. Before this modification, add users' public keys to the server. Edit `/etc/ssh/sshd_config` and add the following lines:
+
+    ChallengeResponseAuthentication no
+    PasswordAuthentication no
+
+Restart the SSH service once again:
+
+    sudo systemctl restart sshd
+
+In addition, the users should be prevented from logging in to the system. This can be achieved by issuing the following command on Ubuntu:
+
+    usermod -s /bin/false user
+
+On RHEL, the corresponding command is:
+
+    usermod -s /sbin/nologin user
+
+The system administrator should also implement a monitoring and alerting system regarding anomalous logins.
+
+## 3 Admin UI (Central Server and Security Server)
+
+### 3.1 Host header injection mitigation
+
+The host header specifies which website or web application should process an incoming HTTP request. The web server uses the value of this header to dispatch the request to the specified website or web application.
+
+By default, this header allows any value which would be a security risk if Admin UI could be accessed by bad actors. To mitigate this issue it suggested to configure `allowed-hosts` as described in [UG-SYSPAR](ug-syspar_x-road_v6_system_parameters.md). 
```

Comment:
It looks like the parameter in the system parameters document is called `allowed-hostnames` rather than `allowed-hosts`.